extensions/wikia/Helios/User.class.php

```diff
@@ -198,6 +198,83 @@ public static function onUserCheckPassword( $id, $username, $hash, $password, &$
 		return true;
 	}
 
+    /**
+     * Called in ExternalUser_Wikia registers a user.
+     *
+     * @param string $username string of the user name
+     * @param string $password string of the plaintext password the user entered
+     * @param string $email string of the user email
+     *
+     * @return boolean true on success, false otherwise
+     */
+    public static function register( $username, $password, $email, $birthdate )
+    {
+        $logger = \Wikia\Logger\WikiaLogger::instance();
+        $logger->info( 'HELIOS_REGISTRATION START', [ 'method' => __METHOD__ ] );
+
+        global $wgHeliosBaseUri, $wgHeliosClientId, $wgHeliosClientSecret;
+        $helios = new Client( $wgHeliosBaseUri, $wgHeliosClientId, $wgHeliosClientSecret );
+
+        try {
+            $registration = $helios->register( $username, $password, $email, $birthdate );
+            $result = !empty( $registration->success );
+
+            if ( !empty( $registration->error ) ) {
+                $logger->error(
+                    'HELIOS_REGISTRATION ERROR_FROM_SERVICE',
+                    [ 'method' => __METHOD__ ]
+                );
+            }
+        }
+
+        catch ( ClientException $e ) {
+            $logger->error(
+                'HELIOS_REGISTRATION ERROR_FROM_CLIENT',
+                [ 'exception' => $e, 'method' => __METHOD__ ]
+            );
+            $result = false;
+        }
+
+        return $result;
+    }
+
+    public static function onRegister( &$result, &$userId, $User, $password, $email ) {
+
+        $heliosUserId = null;
+        $heliosResult = self::register( $User->mName, $password, $email, $User->mBirthDate );
+        $logger = \Wikia\Logger\WikiaLogger::instance();
+
+        global $wgHeliosRegistrationShadowMode;
+
+        if ( $heliosResult ) {
+
+            global $wgExternalSharedDB;
+
+            $table = $wgHeliosRegistrationShadowMode ? '`user`' : 'user_helios';
+
+            $dbw = \wfGetDB( DB_MASTER, [], $wgExternalSharedDB );
+            $heliosUserId = $dbw->selectField( $table, 'user_id', [ 'user_name' => $User->mName ], __METHOD__ );
```

Comment:
Please double-check that the $heliosUserId will be an int and not a string containing number. I'm not sure what is returned from the selectField() here.